packages/grafana-data/src/dataframe/processDataFrame.ts

```diff
@@ -248,22 +248,24 @@ export function guessFieldTypeForField(field: Field): FieldType | undefined {
  * @param guessDefined Whether to guess types of fields with already defined types
  */
 export const guessFieldTypes = (series: DataFrame, guessDefined = false): DataFrame => {
-  for (const field of series.fields) {
-    if (!field.type || field.type === FieldType.other || guessDefined) {
-      // Something is missing a type, return a modified copy
-      return {
-        ...series,
-        fields: series.fields.map(field => {
-          if (field.type && field.type !== FieldType.other && !guessDefined) {
-            return field;
-          }
-          // Calculate a reasonable schema value
-          return {
-            ...field,
-            type: guessFieldTypeForField(field) || FieldType.other,
-          };
-        }),
-      };
+  if (series.fields) {
```

Comment:
Maybe do an early return here instead. Much more readable.

```ts
if(!series.fields){
  return series;
}
```

Reply:
Yep I agree, changed it to make it more readable. Thank you.